### PR TITLE
[HER-72] Change data repository

### PR DESCRIPTION
**Pull request by ConstantinescuAndrei**
## Desctiption
Create "Insert" method

## Related
[HER-72](https://nerds-interns.atlassian.net/jira/software/projects/HER/boards/1?selectedIssue=HER-72)

### CODE REVIEW

File: src/Data/Detection/DetectionRepository.cs [hunk 1]
```diff
@@ -17,14 +17,14 @@ public IEnumerable<MDetection> ByParameter() => _context.Set<EDetection>()
...
             {
-                Class = "glass",
-                Score = 100,
-                Timestamp = DateTime.UtcNow,
-            },
-        };
+                _context.Add(entity);
+                _context.SaveChanges();
+                return entity;
+            })
```
Comment (RaduBuciuceanu) [inline src/Data/Detection/DetectionRepository.cs:27]: Should we extract this lambda into a separate private method?

Comment (RaduBuciuceanu) [reply src/Data/Detection/DetectionRepository.cs:27]: Should we also remove the unused import at the top of the file?

File: test/Data.UnitTests/Detection/DetectionFactoryTests.cs [hunk 2]
```diff
@@ -26,17 +29,32 @@ public void Factory_ShouldConvert_Entity()
             _model.ShouldBeEquivalentTo(outputDetection);
         }
 
+        [Test]
+        public void Factory_shouldConvert_MCreateDetection()
```
Comment (RaduBuciuceanu) [inline test/Data.UnitTests/Detection/DetectionFactoryTests.cs:33]: Should we use the capital letter for the `_shouldConvert_` part?

Comment (RaduBuciuceanu) [reply test/Data.UnitTests/Detection/DetectionFactoryTests.cs:33]: Whould the name `Factory_ShouldConvert_Input` be more relevant here?

File: test/Data.UnitTests/Detection/DetectionFactoryTests.cs [hunk 3]
```diff
@@ -26,17 +29,32 @@ public void Factory_ShouldConvert_Entity()
...
         private static EDetection CreateEntity() => new ()
         {
-            Id = 1,
+            Id = 0,
+            Class = "Class",
+            Score = 100,
+            Timestamp = new DateTime(2021, 07, 20),
+        };
+
+        private static MCreateDetection CreateDetection() => new ()
```
Comment (RaduBuciuceanu) [inline test/Data.UnitTests/Detection/DetectionFactoryTests.cs:48]: Would the name `CreateInput` be more relevant here?

**Review by RaduBuciuceanu — CHANGES_REQUESTED** [review]

File: src/Data/Detection/DetectionRepository.cs (hunk 1 above)
Comment (ConstantinescuAndrei) [reply src/Data/Detection/DetectionRepository.cs]: Done

File: test/Data.UnitTests/Detection/DetectionFactoryTests.cs (hunk 2 above)
Comment (ConstantinescuAndrei) [reply test/Data.UnitTests/Detection/DetectionFactoryTests.cs]: Done

File: test/Data.UnitTests/Detection/DetectionFactoryTests.cs (hunk 3 above)
Comment (ConstantinescuAndrei) [reply test/Data.UnitTests/Detection/DetectionFactoryTests.cs]: Done

**Review by danidoff — APPROVED** [review]

**Review by RaduBuciuceanu — APPROVED** [review]

**Pull request merged**